fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java

```diff
@@ -0,0 +1,102 @@
+import nva.commons.core.paths.UriWrapper;
+import org.joda.time.DateTime;
+
+@SuppressWarnings("PMD")
+
+public class FsApi {
+
+    public static final String FODSELSDATO_IDENTIFIER = "fodselsdato0";
+    public static final String PERSONNUMMER_IDENTIFIER = "personnummer0";
+    public static final String FS_HOST = "api.fellesstudentsystem.no";
+    private static final String PERSON_PATH = "personer";
+    private static final String STUDENTUNDERVISNING_PATH = "studentundervisning";
+    private static final String UNDERVISNING_SEMESTER_AR_PATH = "undervisning.semester.ar";
+    private static final String PERSON_PERSONLOPENUMMER_PATH = "person.personlopenummer";
+    private static final String DB_IDENTIFIER = "dbId";
+    private static final String LIMIT_IDENTIFIER = "limit";
+    private final String username = new Environment().readEnv("FS_USERNAME");
+    private final String password = new Environment().readEnv("FS_PASSWORD");
+    private final HttpClient httpClient;
+    private final String baseFsHostUrl;
+
+    public FsApi (HttpClient httpClient, String baseFsHostUrl) {
+        this.httpClient = httpClient;
+        this.baseFsHostUrl = baseFsHostUrl;
+    }
+
+    @JacocoGenerated
+    public FsApi() {
+        this(HttpClient.newBuilder().build(), new Environment().readEnv("FS_HOST") );
+    }
+
+
+    public FsIdNumber getFsId(FsNin nationalIdentityNumber) throws IOException, InterruptedException {
+        var httpClient = HttpClient.newBuilder().build();
+        var httpRequest = HttpRequest.newBuilder(createSearchPersonUri(nationalIdentityNumber))
+                .header("Authorization", getBasicAuthenticationHeader())
+                .GET().build();
+
+        var response = httpClient.send(httpRequest, HttpResponse.BodyHandlers.ofString());
+        var responseBody = response.body();
+        var fsIdSearchResult = JsonUtils.dtoObjectMapper.readValue(responseBody, FsPersonSearchResponse.class);
+        return fsIdSearchResult.getSearchResults().get(0).getFsPerson().getFsIdNumber();
+    }
+
+    public List<FsCourseData> getCourses(FsIdNumber fsIdNumber) throws IOException, InterruptedException {
+        var httpClient = HttpClient.newBuilder().build();
+        var httpRequest = HttpRequest.newBuilder(createSearchCourseUri(fsIdNumber))
+                .header("Authorization", getBasicAuthenticationHeader())
+                .GET().build();
+
+        var response = httpClient.send(httpRequest, HttpResponse.BodyHandlers.ofString());
+        var responseBody = response.body();
+        var fsCoursesSearchResult = JsonUtils.dtoObjectMapper.readValue(responseBody, FsCoursesSearchResult.class);
+        return fsCoursesSearchResult.getFsCourseData();
+    }
+
+    private URI createSearchPersonUri(FsNin nin) {
+        return UriWrapper.fromUri(baseFsHostUrl)
+                .addChild(PERSON_PATH)
+                .addQueryParameter(DB_IDENTIFIER, "true")
+                .addQueryParameter(LIMIT_IDENTIFIER, "0")
```

Comment:
Rename "0" to constant NO_LIMIT, since that's what it does

Reply:
fixed